include/swift/Sema/ConstraintSystem.h

```diff
@@ -1386,6 +1386,9 @@ enum class ConstraintSystemFlags {
   /// CodeCompletionExpr, and should continue in the presence of errors wherever
   /// possible.
   ForCodeCompletion = 0x40,
+
+  /// Including Clang function types when checking equality for function types.
```

Comment:
This could use some more clarification about what it really means - (if I understand correctly) additional attribute checks while matching function types.

Reply:
How about something like this:

```
  /// Include Clang function types when checking equality for function types.
  ///
  /// When LangOptions.UseClangFunctionTypes is set, we can synthesize
  /// different @convention(c) function types with the same parameter and result
  /// types (similarly for blocks). These types should be treated as semantically
  /// different, as they may have different calling conventions, say due to Clang
  /// attributes such as `__attribute__((ns_consumed))`.
```

Reply:
Great!